src/System.Net.Sockets.Legacy/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -1007,7 +1006,7 @@ private void Accept_Helper(IPAddress listenOn, IPAddress connectTo)
                 SocketClient client = new SocketClient(serverSocket, connectTo, port);
                 Socket clientSocket = serverSocket.Accept();
                 Assert.True(clientSocket.Connected);
-                Assert.True(clientSocket.DualMode);
+                Assert.True((listenOn != IPAddress.IPv6Any && !listenOn.IsIPv4MappedToIPv6) || clientSocket.DualMode);
```

Comment:
Should we split this test in two, one for Unix and one for Windows, so that we can detect if, for example, the Windows behavior "regresses" to that of Unix?